### PR TITLE
Reorganize MOL file

### PR DESCRIPTION
**Pull request by marchdf**
This should cause zero diffs because it is just a reorg. It brings relevant things closer together so that the next step is easier. I also rewrote a bunch of comments

### CODE REVIEW

File: Source/Diffusion.cpp [hunk 1]
```diff
@@ -73,14 +70,13 @@ PeleC::getMOLSrcTerm(
...
+  // BCs
   const amrex::StateDescriptor* desc = state[State_Type].descriptor();
   const auto& bcs = desc->getBCs();
   amrex::Gpu::DeviceVector<amrex::BCRec> d_bcs(desc->nComp());
   amrex::Gpu::copy(
     amrex::Gpu::hostToDevice, bcs.begin(), bcs.end(), d_bcs.begin());
 
-  // Fetch some gpu arrays
   prefetchToDevice(S);
   prefetchToDevice(MOLSrcTerm);
```
Comment (jrood-nrel) [inline Source/Diffusion.cpp]: I never see these in other applications. Do we know if they even do anything?

Comment (marchdf) [reply Source/Diffusion.cpp]: Yeah I have no idea. I think this is a relic. It also is only used for managed mem: https://github.com/AMReX-Codes/amrex/blob/96b811d2cb9f24e87428e4ebc9904b4dcdf73e7f/Src/Base/AMReX_BaseFab.H#L1689. 

So I will just remove these.

File: Source/Diffusion.cpp [hunk 2]
```diff
@@ -31,15 +31,16 @@ PeleC::getMOLSrcTerm(
```
Comment (baperry2) [inline Source/Diffusion.cpp:29]: I think the part about face-centering in steps 1-2 above is not accurate, can that be updated

File: Source/Diffusion.cpp [hunk 3]
```diff
@@ -73,14 +70,13 @@ PeleC::getMOLSrcTerm(
   const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> dxD = {
     {AMREX_D_DECL(dx1, dx1, dx1)}};
 
-  // Grab the BCs
+  // BCs
```
Comment (baperry2) [inline Source/Diffusion.cpp]: I think these are only used for EB redistribution so this could be moved down to the EB section

**Review by baperry2 — APPROVED** [review]

File: Source/Diffusion.cpp (hunk 2 above)
Comment (marchdf) [reply Source/Diffusion.cpp:29]: good catch, updated

File: Source/Diffusion.cpp (hunk 3 above)
Comment (marchdf) [reply Source/Diffusion.cpp]: another good catch, moved them down to that section

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**